python/dgl/readout.py

```diff
@@ -417,8 +417,9 @@ def broadcast_nodes(graph, graph_feat, *, ntype=None):
     broadcast_edges
     """
     if (F.shape(graph_feat)[0] != graph.batch_size and graph.batch_size == 1):
-        dgl_warning('If single graph, please make sure the batching dimension is added'
-                    ' and the tensor shape of graph_feat is (1, *).')
+        dgl_warning(':func:`broadcast_nodes` directly taking one node feature tensor as `feat[i]`'
+                    ' to broadcast to all nodes in a single graph will be deprecated in future,'
+                    ' please make sure the feature tensor shape is :math:`(1, *).`')
```

Comment:
Perhaps you can say

"""
For a single graph, use a tensor of shape (1, *) for graph_feat. The support of shape (*) will be deprecated.
"""

Reply:
Thank you for you suggestion. It's more concise.